pallene/driver.lua

```diff
@@ -83,26 +81,27 @@ function driver.compile_internal(filename, stop_after)
 end
 
 local function compile_pallene_to_c(pallene_filename, c_filename, mod_name)
-    local ok, err, errs
-
-    local module
-    module, errs = driver.compile_internal(pallene_filename)
-    if not module then
-        return false, errs
-    end
-
-    local c_code
-    c_code, errs = coder.generate(module, mod_name)
-    if not c_code then
-        return c_code, errs
-    end
-
-    ok, err = util.set_file_contents(c_filename, c_code)
-    if not ok then
-        return ok, {err}
+    local ok, err, errs, input = false, nil, {}, nil
```

Comment:
I think the old version of this function was easier to read than the new one. 

1. Please use the early exit pattern instead of nested if statements. `if not blah then return blah, errs`. This way the program reads as a linear sequence instead of a big triangle.
2. Please declare the variables close to where they are used instead of all at the top of the function.
3. We don't need to initialize the variables in the foward declaration, since the value is going to be overwritten anyway.